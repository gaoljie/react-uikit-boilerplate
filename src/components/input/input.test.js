import React from "react";
import "jest-styled-components";
import { mount, shallow } from "enzyme";

import Input from "./index";

const props = {
  type: "email",
  name: "email",
  value: "aa@bb.com",
  width: "200px",
  placeholder: "Please Input Email"
};

describe("Test Input", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Render Prop", () => {
    const wrapper = shallow(<Input {...props} />);

    Object.keys(props).forEach(key => {
      expect(wrapper.props()[key]).toEqual(props[key]);
    });
  });

  it("Event Handler", () => {
    const wrapper = shallow(
      <Input
        {...props}
        onChange={e => wrapper.setProps({ value: e.target.value })}
      />
    );

    const newValue = "newValue";

    wrapper.simulate("change", {
      target: {
        value: "newValue"
      }
    });

    expect(wrapper.props().value).toEqual(newValue);
  });
});
