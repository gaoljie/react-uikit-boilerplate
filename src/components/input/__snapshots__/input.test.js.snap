// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Matches the snapshot 1`] = `
.c0 {
  width: 200px;
  border: 1px solid #EF6F35;
  box-sizing: border-box;
  padding-left: 16px;
  height: 38px;
  font-size: 16px;
  outline: none;
}

<Input
  name="email"
  placeholder="Please Input Email"
  type="email"
  value="aa@bb.com"
  width="200px"
>
  <Styled(styled.div)
    as="input"
    name="email"
    placeholder="Please Input Email"
    type="email"
    value="aa@bb.com"
    width="200px"
  >
    <StyledComponent
      as="input"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
              "
  border: 1px solid ",
              "#EF6F35",
              ";
  box-sizing: border-box;
  padding-left: 16px;
  height: 38px;
  font-size: 16px;
  outline: none;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      name="email"
      placeholder="Please Input Email"
      type="email"
      value="aa@bb.com"
      width="200px"
    >
      <input
        className="c0"
        name="email"
        placeholder="Please Input Email"
        type="email"
        value="aa@bb.com"
        width="200px"
      />
    </StyledComponent>
  </Styled(styled.div)>
</Input>
`;
